code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1325,7 +1325,63 @@
 		if(I)
 			kept_items[I] = thing
 
-	dna.species.create_organs(src)
+	if(retain_damage)
+		//Create a list of body parts which are damaged by burn or brute and save them to apply after new organs are generated. First we just handle external organs.
+		var/bodypart_damages = list()
+		//Loop through all external organs and save the damage states for brute and burn
+		for(var/obj/item/organ/external/E in bodyparts)
+			if(E.brute_dam == 0 && E.burn_dam == 0) //If there's no damage we don't bother remembering it.
+				continue
+			var/brute = E.brute_dam
+			var/burn = E.burn_dam
+			var/IB = E.internal_bleeding
+			var/obj/item/organ/external/OE = new E.type()
+			var/stats = list(OE, brute, burn, IB)
+			bodypart_damages += list(stats)
+
+		//Now we do the same for internal organs via the same proceedure.
+		var/internal_damages = list()
+		for(var/obj/item/organ/internal/I in internal_organs)
+			if(I.damage == 0)
+				continue
+			var/obj/item/organ/internal/OI = new I.type()
+			var/damage = I.damage
+			var/stats = list(OI, damage)
+			internal_damages += list(stats)
+
+		//Create the new organs for the species change
+		dna.species.create_organs(src)
+
+		//Apply relevant damages and variables to the new organs.
+		for(var/obj/item/organ/external/E in bodyparts)
+			for(var/list/part in bodypart_damages)
+				var/obj/item/organ/external/OE = part[1]
+				if((E.type == OE.type)) // Type has to be explicit, as right limbs are a child of left ones etc.
+					var/brute = part[2]
+					var/burn = part[3]
+					var/IB = part[4]
+					//Deal the damage to the new organ and then delete the entry to prevent duplicate checks
+					E.receive_damage(brute, burn, ignore_resists = TRUE)
+					E.internal_bleeding = IB
+					qdel(part)
+
+		for(var/obj/item/organ/internal/I in internal_organs)
+			for(var/list/part in internal_damages)
+				var/obj/item/organ/internal/OI = part[1]
+				var/organ_type
+
+				if(OI.parent_type == /obj/item/organ/internal) //Dealing with species organs

```

Comment:
Species organs are sibling classes, I wasn't sure if there was already a way to check this.